aibolit/patterns/classic_setter/classic_setter.py

```diff
@@ -11,24 +8,38 @@ class ClassicSetter:
     def __init__(self):
         pass
 
-    def value(self, filename: str) -> List[LineNumber]:
-        lst: List[LineNumber] = []
-        tree = build_ast(filename).filter(javalang.tree.MethodDeclaration)
-        for _, node in tree:
-            if (node.return_type is None) and ('set' in node.name[:3]):
-                if (isinstance(node.body, list)) and len(node.body) < 2:
-                    for statement in node.body:
-                        if isinstance(statement, javalang.tree.StatementExpression):
-                            if isinstance(statement.expression, javalang.tree.Assignment):
-                                expression = statement.expression.expressionl
-                                if isinstance(expression, javalang.tree.This):
-                                    if statement.expression.type == '=':
-                                        if expression.selectors[0].member.lower() == node.name.lower()[3:]:
-                                            lst.append(node._position.line)
-                                    else:
-                                        break
-                                else:
-                                    break
-                            else:
-                                break
-        return lst
+    def check_nodes(self, ast: AST, body_nodes: List) -> bool:
+        '''
+        In this function we check whether
+        nodes in the function are agree with
+        the definition of classic setter or not.
+        Classic setter should consist of only several
+        assert_statement and THIS statement.
+        This statement is not-directed child of
+        STATEMENT_EXPRESSION.
+        '''
+        suitable_nodes: List[ASTNodeType] = [
+            ASTNodeType.ASSERT_STATEMENT,
+            ASTNodeType.STATEMENT_EXPRESSION,
+        ]
+        for node in body_nodes:
+            if node.node_type not in suitable_nodes:
+                return False
+        return True
+
+    def value(self, filename: str) -> List[int]:
+        lst: Set[int] = set()
+        ast = AST.build_from_javalang(build_ast(filename))
+        method_decls = list(ast.get_proxy_nodes(ASTNodeType.METHOD_DECLARATION))
+        for node in method_decls:
+            method_name = node.name
+            if node.return_type is None:
+                if method_name.startswith('set') and self.check_nodes(ast, node.body):
+                    for child_this in ast.get_proxy_nodes(ASTNodeType.THIS):
+                        child_membref = child_this.selectors
+                        if len(child_membref):
```

Comment:
I've got lost. It is probably needed to be extracted into separate method with some nice name.